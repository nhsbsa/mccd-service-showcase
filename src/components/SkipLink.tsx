import React from 'react';
import Link from 'next/link';

type SkipProps = {
  toId: string
  text?: string
}
function SkipLink({ toId = '', text = 'Skip to main content' }:SkipProps) {
  return (<Link scroll href={`#${toId}`} className="govuk-skip-link">{text}</Link>);
}

export default SkipLink;
