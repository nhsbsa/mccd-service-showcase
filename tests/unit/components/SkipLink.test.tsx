import React from 'react';
import { render, screen } from '@testing-library/react';
import SkipLink from '@/components/SkipLink';

describe('SkipLink component', () => {
  it('renders without crashing', () => {
    render(<SkipLink toId="main-content" />);
    expect(screen.getByText('Skip to main content')).toBeInTheDocument();
  });

  it('renders with custom text when provided', () => {
    render(<SkipLink toId="main-content" text="Skip to content" />);
    expect(screen.getByText('Skip to content')).toBeInTheDocument();
  });

  it('sets the href attribute correctly', () => {
    render(<SkipLink toId="main-content" />);
    const link = screen.getByRole('link');
    expect(link).toHaveAttribute('href', '#main-content');
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<SkipLink toId="main-content" text="Skip to content" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
