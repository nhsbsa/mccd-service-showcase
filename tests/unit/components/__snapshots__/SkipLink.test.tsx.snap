// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipLink component matches the snapshot 1`] = `
<DocumentFragment>
  <a
    class="govuk-skip-link"
    href="#main-content"
  >
    Skip to content
  </a>
</DocumentFragment>
`;
